### PR TITLE
Update self-managed templates to use internal LB for node-to-node communication

### PR DESCRIPTION
**Pull request by nawazkh**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
- This PR aims to update self-managed templates hosted by CAPZ. 
- The self managed templates will now use the Internal LB for node-node communication.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

- Next will update managed K8s templates.

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Update self-managed templates to use internal LB for node-to-node communication
```

### CODE REVIEW

**Comment by nawazkh** [comment]
> Why don't we want the changes from #5209 for these templates?

I am not sure at the moment if we want #5209 changes for these templates(`ephemeral`,`bastion`,`azure-cni-v1` and `edgezone`) exclusively. 
Earlier these four flavors treated `default` template as its `base` flavor. That doesn't seem right. This PR cleans up that approach of building each flavor without creating a dependence on `default` template.

**Comment by nojnhuh** [comment]
Do these templates produce nodes that can communicate with the control plane without the fix from #5209?

**Comment by nawazkh** [comment]
> Do these templates produce nodes that can communicate with the control plane without the fix from #5209?

I don't think so. #5209 is still WIP.
But even if they do, keeping these four flavors independent of `default` and relying on `base` to scaffold themselves seems logical to me.

**Comment by nojnhuh** [comment]
I think that coupling with the default template is intended exactly for the kinds of changes we're making to look here. With the approach in this PR, we'll now need to copy-paste those same patches defined for the default template in #5209 to these four other templates, vs. defining them once in the default template and letting them propagate to these other templates automatically.

**Comment by nawazkh** [comment]
A change like #5209 needs to be propagated across all templates and not just to these four flavors. 
This PR just cleans up the way each flavor is built.

**Comment by nawazkh** [comment]
WIP, hold.
/hold

**Comment by nawazkh** [comment]
I.. closed this PR? How?
/reopen

**Review by willie-yao — APPROVED** [review]
/lgtm

Thanks for your hard work on this @nawazkh! 🚀

File: templates/cluster-template-edgezone.yaml
Comment (mboersma) [inline templates/cluster-template-edgezone.yaml:1]: Did we intend to delete the edgezone template? Looks like it's still there and been updated--maybe it didn't generate successfully?

Comment (nawazkh) [reply templates/cluster-template-edgezone.yaml:1]: ... this should not be deleted. 
/hold

Comment (nawazkh) [reply templates/cluster-template-edgezone.yaml:1]: added via 987cb4f9cc22ace9ad0ee7e17962eeecac67ba22

Comment (nawazkh) [reply templates/cluster-template-edgezone.yaml:1]: /unhold

**Comment by willie-yao** [comment]
/lgtm

Good catch @mboersma!

**Review by mboersma — APPROVED** [review]
/lgtm

**Pull request merged**